src/System.Management.Automation/engine/Attributes.cs

```diff
@@ -1359,7 +1360,7 @@ public ValidateCountAttribute(int minLength, int maxLength)
     [AttributeUsage(AttributeTargets.Field | AttributeTargets.Property)]
     public sealed class ValidateSetAttribute : ValidateEnumeratedArgumentsAttribute
     {
-        private string[] _validValues;
```

Comment:
I think this shouldn't change, and I would add a new read-only property for the Type so you can tell the difference on which constructor was used.